daemon/daemon_unix.go

```diff
@@ -1757,7 +1757,23 @@ func (daemon *Daemon) setupSeccompProfile() error {
 }
 
 func (daemon *Daemon) useShimV2() bool {
-	return cgroups.IsCgroup2UnifiedMode()
+	env := os.Getenv("MOBY_DISABLE_SHIM_V2")
+	if env == "" {
+		return true
+	}
+	// MOBY_DISABLE_SHIM_V2 is deprecated from its birth.
+	// MOBY_DISABLE_SHIM_V2 is just added as a workaround for potential regression issues.
+	logrus.Warn("MOBY_DISABLE_SHIM_V2 is deprecated and planned to be removed in future releases")
+	disableV2, err := strconv.ParseBool(env)
+	if err != nil {
+		logrus.Warnf("Ignoring invalid MOBY_DISABLE_SHIM_V2 value %q: %v", env, err)
+		return true
+	}
+	if disableV2 && cgroups.IsCgroup2UnifiedMode() {
+		logrus.Warn("MOBY_DISABLE_SHIM_V2 is not supported on cgroup v2 hosts. Ignoring.")
```

Comment:
Should we make this a hard failure instead of logging?

Reply:
updated